### ISSUE TITLE
Update Vercel adapter to Node 14.8 to fix top level await builds

### ISSUE THREAD

**Issue by digitaldeus** [sveltejs/kit#4782]
### Describe the bug

SvelteKit has been updated to `node14.8` to enable top-level await per #4742 .  Would be great if we could update the [vercel adapter](https://github.com/sveltejs/kit/blob/master/packages/adapter-vercel/index.js) to the same version since it is currently forcing build version `node14` and there doesn't appear to be a way to override it

### Reproduction

repo [here](https://github.com/digitaldeus/vercel-top-level-await)

to repro just run:
```
npm install
npm run build
```

### Logs

```shell
> Using @sveltejs/adapter-vercel
✘ [ERROR] Top-level await is not available in the configured target environment ("node14")

    .svelte-kit/output/server/chunks/hooks-45a47dbd.js:1:0:
      1 │ await Promise.resolve(true);
        ╵ ~~~~~

> Build failed with 1 error:
.svelte-kit/output/server/chunks/hooks-45a47dbd.js:1:0: ERROR: Top-level await is not available in the configured target environment ("node14")
.svelte-kit/output/server/chunks/hooks-45a47dbd.js:1:0: ERROR: Top-level await is not available in the configured target environment ("node14")
    at failureErrorWithLog (/Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:1603:15)
    at /Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:1249:28
    at runOnEndCallbacks (/Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:1162:65)
    at buildResponseToResult (/Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:1247:7)
    at /Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:1356:14
    at /Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:666:9
    at handleIncomingPacket (/Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:763:9)
    at Socket.readFromStdout (/Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:632:7)
    at Socket.emit (node:events:527:28)
    at addChunk (node:internal/streams/readable:324:12)
```

### System Info

```shell
System:
    OS: macOS 12.3.1
    CPU: (8) arm64 Apple M1
    Memory: 96.47 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.8.0 - /opt/homebrew/bin/node
    Yarn: 1.22.18 - ~/.yarn/bin/yarn
    npm: 8.5.5 - /opt/homebrew/bin/npm
  Browsers:
    Brave Browser: 100.1.37.116
    Chrome: 100.0.4896.127
    Safari: 15.4
```

### Severity

blocking all usage of SvelteKit

### Additional Information

Unable to use sveltekit in vercel due to top level await issue

**Comment by vhscom** [comment]
Vercel is tied to AWS and Amazon hasn't updated their serverless lambda Node environment to v16 yet (though they have publicly stated May amidst much community discord). There are some janky workarounds but I'd just wait until Vercel is capable of updating which I'd wager they're practically finished with and just waiting on AWS to button it up.

Tracking info here: https://github.com/vercel/community/discussions/37.

Aside: I think AWS has a separate ticket related to being able to specify minor Node versions but don't quote me on that.

**Comment by antony** [comment]
I might suggest that we instead read the node version from the application's config, rather than just update the hardcoded version inside the adapter.

**Comment by digitaldeus** [comment]
Vercel now support [Node16](https://vercel.com/changelog/node-js-16-lts-is-now-available) so allowing the adapter to be dynamic would be a great approach
